src/lib/optimizer/optimizer.cpp

```diff
@@ -105,6 +106,8 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
   // for now we want the PredicateReorderingRule to have the final say on predicate positions
   optimizer->add_rule(std::make_unique<PredicatePlacementRule>());
 
+  optimizer->add_rule(std::make_unique<OrToUnionRule>());
```

Comment:
Did you consider different positions for this rule?

Reply:
Any position between the `ExpressionReductionRule` and the `SubqueryToJoinRule` works. Now, I moved it directly beneath the `PredicateSplitUpRule` to reflect the similarity of these rules.

Reply:
The downside of this order is that the introduced unions remain in the LQP even if no other rule benefits from them. @mrks If I remember correctly, you mentioned that there should be an additional clean-up rule at the end of the optimizer?

Reply:
Not sure if I can follow. Should we discuss the position of the rule and a possible merge of the rules in person?